From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Mon, 24 Apr 2023 17:05:31 +0200
Subject: [PATCH] Engine: Remove static_assert on pthread_t

In x86_64 Serenity, `sizeof(size_t) == 8` and `sizeof(pthread_t) == 4`.
The game seems to run just fine without this assertion, and changing
`pthread_t` is quite the yak hole.
---
 SamTFE/Sources/Engine/Base/Unix/UnixSynchronization.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/SamTFE/Sources/Engine/Base/Unix/UnixSynchronization.cpp b/SamTFE/Sources/Engine/Base/Unix/UnixSynchronization.cpp
index cfa9eda4de2a00e44ec4c8949e718740776656ef..74d1908c3d3dd6264c7c27ce7d7a2c704462b1ed 100644
--- a/SamTFE/Sources/Engine/Base/Unix/UnixSynchronization.cpp
+++ b/SamTFE/Sources/Engine/Base/Unix/UnixSynchronization.cpp
@@ -58,7 +58,6 @@ LONG InterlockedDecrement(LONG volatile *Addend) {
 }
 
 unsigned long long GetCurrentThreadId() {
-  static_assert(sizeof(pthread_t) == sizeof(size_t), "");
   return (unsigned long long) pthread_self();
 }
 
